src/Microsoft.DotNet.ProjectJsonMigration/transforms/TransformApplicator.cs

```diff
@@ -0,0 +1,170 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Build.Evaluation;
+using Microsoft.Build.Construction;
+using Microsoft.DotNet.ProjectModel;
+using Microsoft.DotNet.Cli.Utils;
+using Microsoft.DotNet.Cli;
+using System.Linq;
+using System.IO;
+using Newtonsoft.Json;
+
+namespace Microsoft.DotNet.ProjectJsonMigration
+{
+    public class TransformApplicator : ITransformApplicator
+    {
+        private readonly ProjectRootElement _projectElementGenerator = ProjectRootElement.Create();
+
+        public void Execute<T, U>(
+            T element,
+            U destinationElement) where T : ProjectElement where U : ProjectElementContainer
+        {
+            if (element != null)
+            {
+                if (typeof(T) == typeof(ProjectItemElement))
```

Comment:
I was thinking along these lines to have something like a composite, that routes the appropriate applicator. This would be nice for testing the individual applicators too.